### ISSUE TITLE
[atomic_rmw]: Support for fp16

### ISSUE THREAD

**Issue by etiotto** [intel/intel-xpu-backend-for-triton#152]
The current implementation of `tt.atomic_rmw` is functional for several data types but, crucially, it doesn't yet work for `float16`.
The issue is in the LLVM-SPIR translator

**Comment by whitneywhtsang** [comment]
IGC JIRA: https://jira.devtools.intel.com/browse/GSD-6841

**Comment by aregm** [comment]
@etiotto "The feature is now implemented in the llvm-spirv translator" - is this still missing?

**Comment by whitneywhtsang** [comment]
> @etiotto "The feature is now implemented in the llvm-spirv translator" - is this still missing?

The llvm-spirv translator support is implemented, but the IGC support is still missing.

**Comment by etiotto** [comment]
> > @etiotto "The feature is now implemented in the llvm-spirv translator" - is this still missing?
> 
> The llvm-spirv translator support is implemented, but the IGC support is still missing.

Yep. The IGC work is tracked in Jira https://jira.devtools.intel.com/browse/GSD-6841

**Comment by etiotto** [comment]
The PVC GPU HW doesn't support atomic operations on fp16 (and bf16) data types.  Currently the team is working on emulating support in Triton as an *experimental* feature. Performance is expected to be poor. 

We will reconsider supporting this feature once we have HW support.

**Comment by vlad-penkin** [comment]
Closing the issue based on the @etiotto comment.
